datafusion/physical-expr/src/utils.rs

```diff
@@ -1972,4 +1977,55 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn project_empty_output_ordering() -> Result<()> {
+        let schema = Schema::new(vec![
+            Field::new("a", DataType::Int32, true),
+            Field::new("b", DataType::Int32, true),
+            Field::new("c", DataType::Int32, true),
+        ]);
+        let empty_ordering = find_orderings_of_exprs(
+            &[
+                (Arc::new(Column::new("b", 1)), "b_new".to_string()),
+                (Arc::new(Column::new("a", 0)), "a_new".to_string()),
+            ],
+            Some(&[PhysicalSortExpr {
+                expr: Arc::new(Column::new("b", 1)),
+                options: SortOptions::default(),
+            }]),
+            EquivalenceProperties::new(Arc::new(schema.clone())),
+            OrderingEquivalenceProperties::new(Arc::new(schema.clone())),
+        )?;
+
+        assert_eq!(
+            vec![
+                Some(PhysicalSortExpr {
+                    expr: Arc::new(Column::new("b_new", 0)),
+                    options: SortOptions::default(),
+                }),
+                None,
+            ],
+            empty_ordering
```

Comment:
nit: it doesn't look like empty.

```suggestion
        let one_ordering = find_orderings_of_exprs(
            &[
                (Arc::new(Column::new("b", 1)), "b_new".to_string()),
                (Arc::new(Column::new("a", 0)), "a_new".to_string()),
            ],
            Some(&[PhysicalSortExpr {
                expr: Arc::new(Column::new("b", 1)),
                options: SortOptions::default(),
            }]),
            EquivalenceProperties::new(Arc::new(schema.clone())),
            OrderingEquivalenceProperties::new(Arc::new(schema.clone())),
        )?;

        assert_eq!(
            vec![
                Some(PhysicalSortExpr {
                    expr: Arc::new(Column::new("b_new", 0)),
                    options: SortOptions::default(),
                }),
                None,
            ],
            one_ordering
```